nomad/structs/config/vault.go

```diff
@@ -245,9 +245,15 @@ func (c *VaultConfig) Equals(b *VaultConfig) bool {
 	if c.TLSKeyFile != b.TLSKeyFile {
 		return false
 	}
-	if c.TLSSkipVerify != b.TLSSkipVerify {
+
+	if c.TLSSkipVerify == nil || b.TLSSkipVerify == nil {
+		if c.TLSSkipVerify != b.TLSSkipVerify {
+			return false
+		}
+	} else if *c.TLSSkipVerify != *b.TLSSkipVerify {
```

Comment:
i was thinking, maybe we could have `pointer.Compare` similar to (and replacing) `helper.CompareTimePtrs` - caveat being it should only be used on basic types (not on structs with also pointer fields)

Reply:
Good idea, lemme take a crack at that.

Reply:
> caveat being it should only be used on basic types

what am I thinking, we can enforce that with a type parameter!

Reply:
pretty sure this is effectively what we'd want

```
// Compare returns whether a and b contain the same value.
func Compare[A constraints.Ordered](a, b *A) bool {
	return *a == *b
}```

Reply:
Yeah, something like this? https://go.dev/play/p/iZx7yLMGIoU

Reply:
the problem is I want to prevent comparisons like this:

https://go.dev/play/p/Lh26c-YWmBd

using `constraints.Ordered` as a constraint prevents comparing pointers of anything but underlying Numeric/String types which are safe

Reply:
hrm, I seem to be stuck there.

```go
type primitivePointer interface {
	*bool | *string | *int // etc, we can add the rest here
}

// Compare is an equality check for pointers that point to primitives.
func Compare[A primitivePointer](a, b A) bool {
	if a == nil || b == nil {
		return a == b
	}

	return *a == *b
}
```

Gives me:

```
# github.com/hashicorp/nomad/helper/pointer
helper/pointer/pointer.go:28:10: invalid operation: pointers of a (variable of type A constrained by primitivePointer) must have identical base types
helper/pointer/pointer.go:28:16: invalid operation: pointers of b (variable of type A constrained by primitivePointer) must have identical base types
```

For now I'm going to merge the bugfix and then we'll come up with some nicer comparison functions in a separate PR.

Reply:
Ah, I posted that without seeing the other solutions. In any case, probably better for another PR.